meshio/vtu/_vtu.py

```diff
@@ -43,16 +43,96 @@ def _cells_from_data(connectivity, offsets, types, cell_data_raw):
         [[0], numpy.where(types[:-1] != types[1:])[0] + 1, [len(types)]]
     )
 
+    # boolean needed for polyhedron cells
+    found_polyhedron_faces_before = False
+
     cells = []
     cell_data = {}
+    polyhedron_faces = {}
     for start, end in zip(b[:-1], b[1:]):
         try:
             meshio_type = vtk_to_meshio_type[types[start]]
         except KeyError:
             raise ReadError("File contains cells that meshio cannot handle.")
-        if meshio_type == "polygon" or meshio_type == "polyhedron":
+        if meshio_type in ["polygon", "polyhedron"]:
             # Polygons and polyhedra have unknown and varying number of nodes per cell.
 
+            if meshio_type == "polyhedron":
+                # The data format for face-cells is:
+                # num_faces_cell_0,
+                #   num_nodes_face_0, node_ind_0, node_ind_1, ..
+                #   num_nodes_face_1, node_ind_0, node_ind_1, ..
+                #   ...
+                # num_faces_cell_1,
+                #   ...
+                #
+                # The faceoffsets describes the end of the face description for each
```

Comment:
Is faceoffsets a variable? I don't see it anywhere here.

Reply:
It is part of the vtu data format. The comment was more useful a few lines down, so I'll move it.